text/0000-grid-primitives.md

```diff
@@ -0,0 +1,761 @@
+represented as signed integers indicating millionths of degrees.
+
+### Datetime
+
+Datetime values are represented as an
+[ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) combined date-time formats.
+For example, "2007-04-05T14:30Z" or "2007-04-05T12:30-02:00".
+
+## Schema
+
+Property definitions are collected into a Schema data type, which defines all
+the possible properties for an item that belongs to a given Schema. A Schema
+includes the following:
+
+- *name* - the name and identifier for the Schema
+- *description* - description of the Schema
+- *owner* - the Pike compatible org identifier for the owning org of the Schema
+- *properties* - a list of `PropertyDefinitions` that define the attributes of
+  the entity being defined by the Schema
+
+Schemas are used to validate a collection of property values defined in a data
+structure.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+A schema and its property definitions are stored in state in the protobuf
+message format.  These messages are defined as follows:
+
+```
+message PropertyDefinition {
+    enum DataType {
+        UNSET_DATA_TYPE = 0;
+        BYTES = 1;
+        BOOLEAN = 2;
+        NUMBER = 3;
+        STRING = 4;
+        ENUM = 5;
+        STRUCT = 6;
+        LAT_LONG = 7;
+        DATETIME = 8;
+    }
+
+    // The name of the property
+    string name = 1;
+    // The data type of the value; must not be set to UNSET_DATA_TYPE.
+    DataType data_type = 2;
+    // Indicates that this is a required property in the Schema
+    bool required = 3;
+    // An optional description of the field.
+    string description = 4;
+
+    // The exponent for a NUMBER property
+    sint32 number_exponent = 10;
+    // The list of values for an ENUM property; must not be empty/ for
+    // properties of that type.
+    repeated string enum_options = 11;
+    // The list of property definitions for a STRUCT property; must  not be
+    // empty for properties of that type.
+    repeated PropertyDefinition struct_properties = 12;
+}
+
+message Schema {
+    // The name of the Schema.  This is also the unique identifier for the
+    // Schema.
+    string name = 1;
+    // An optional description of the schema.
+    string description = 2;
+    // The Pike organization that has rights to modify the schema.
+    string owner = 3;
+
+    // The property definitions that make up the Schema; must not be empty.
+    repeated PropertyDefinition properties = 10;
+}
+
+// A SchemaList is used to mitigate hash collisions.
+message SchemaList {
+    repeated Schema schemas = 1;
+}
+
+message LatLong {
+      // Coordinates are expected to be in millionths of a degree
+      sint64 latitude = 1;
+      sint64 longitude = 2;
+}
+
+message PropertyValue {
+    // The name of the property value.  Used to validate the property against a
+    // Schema.
+    string name = 1;
+    // The data type of the property.  Indicates which value field the actual
```

Comment:
Should this be "Indicates in which value field ..."?

Reply:
fixed